sdk/test/trace/always_on_sampler_test.cc

```diff
@@ -0,0 +1,43 @@
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/sdk/trace/samplers/always_on.h"
+
+#include <gtest/gtest.h>
+#include <map>
+
+using namespace opentelemetry::sdk::trace;
+using namespace opentelemetry::nostd;
+
+TEST(AlwaysOnSampler, ShouldSample)
+{
+  AlwaysOnSampler sampler;
+
+  // A buffer of trace_id with no specific meaning
+  constexpr uint8_t buf[] = {0, 1, 2, 3, 4, 5, 6, 7, 0, 1, 2, 3, 4, 5, 6, 7};
+
+  trace_api::TraceId trace_id_invalid;
+  trace_api::TraceId trace_id_valid(buf);
+  std::map<std::string, int> key_value_container = {{"key", 0}};
+
+  // Test with invalid (empty) trace id and empty parent context
+  auto sampling_result = sampler.ShouldSample(
+      nullptr, trace_id_invalid, "invalid trace id test", trace_api::SpanKind::kServer,
+      trace_api::KeyValueIterableView<std::map<std::string, int>>(key_value_container));
+
+  ASSERT_EQ(Decision::RECORD_AND_SAMPLE, sampling_result.decision);
+  ASSERT_EQ(nullptr, sampling_result.attributes);
```

Comment:
The behavior between invalid and valid trace ID are the same. Is that intentional?

Reply:
Yes. Because the sampler always returns RECORD_AND_SAMPLED no matter what input it receives.